crates/context_aware_config/src/api/dimension/handlers.rs

```diff
@@ -86,39 +93,155 @@ async fn create(
+    }
+
+    let fun_name = match update_req.function_name {
+        Some(Value::String(func_name)) => Some(func_name),
+        Some(Value::Null) | None => None,
+        _ => {
+            log::error!("Expected a string or null as the function name.");
+            return Err(bad_argument!(
+                "Expected a string or null as the function name."
+            ));
         }
-        Err(e) => {
-            log::error!("Dimension upsert failed with error: {e}");
-            Err(unexpected_error!(
-                "Something went wrong, failed to create/update dimension"
+    };
+    dimension_row.function_name = fun_name.clone();
+
+    if let Some(position_val) = update_req.position.clone() {
+        let new_position: i32 = position_val.clone().into();
+        validate_dimension_position(path.into_inner(), position_val)?;
+
+        let previous_position = dimension_row.position.clone();
+        conn.transaction::<_, superposition::AppError, _>(|transaction_conn| {
+            if previous_position < new_position {
+                diesel::update(dsl::dimensions)
+                    .filter(dimensions::position.gt(dimension_row.position))
+                    .filter(dimensions::position.le(new_position.clone()))
+                    .set((
+                        dsl::last_modified_at.eq(Utc::now().naive_utc()),
+                        dsl::last_modified_by.eq(user.get_email()),
+                        dimensions::position.eq(dimensions::position - 1),
+                    ))
+                    .execute(transaction_conn)?
+            } else {
+                diesel::update(dsl::dimensions)
+                    .filter(dimensions::position.lt(dimension_row.position))
+                    .filter(dimensions::position.ge(new_position.clone()))
+                    .set((
+                        dsl::last_modified_at.eq(Utc::now().naive_utc()),
+                        dsl::last_modified_by.eq(user.get_email()),
+                        dimensions::position.eq(dimensions::position + 1),
+                    ))
+                    .execute(transaction_conn)?
+            };
+
+            let result = diesel::update(dimensions)
+                .filter(dsl::dimension.eq(&dimension_row.dimension))
+                .set((
+                    dsl::last_modified_at.eq(Utc::now().naive_utc()),
+                    dsl::last_modified_by.eq(user.get_email()),
+                    dimensions::function_name.eq(dimension_row.function_name),
+                    dimensions::schema.eq(dimension_row.schema),
+                    dimensions::position.eq(new_position),
+                ))
+                .get_result::<Dimension>(transaction_conn)?;
+            let is_mandatory = tenant_config
+                .mandatory_dimensions
+                .contains(&result.dimension);
+            Ok(HttpResponse::Created()
+                .json(DimensionWithMandatory::new(result, is_mandatory)))
```

Comment:
why is the response `Created` here ?
we are just updating in this api, create should not be possible here